source/lac/scalapack.cc

```diff
@@ -756,6 +866,412 @@ NumberType ScaLAPACKMatrix<NumberType>::norm(const char type) const
   return res;
 }
 
+
+
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::save(const char *filename) const
+{
+#ifndef DEAL_II_WITH_HDF5
+  (void)filename;
+  AssertThrow(false, ExcMessage ("HDF5 support is disabled."));
+#else
+#  ifdef H5_HAVE_PARALLEL
+  //implementation for configurations equipped with a parallel file system
+  save_parallel(filename);
+
+#  else
+  //implementation for configurations with no parallel file system
+  save_serial(filename);
+
+#  endif
+#endif
+}
+
+
+
+template <typename NumberType>
+void ScaLAPACKMatrix<NumberType>::save_serial(const char *filename) const
+{
+#  ifndef DEAL_II_WITH_HDF5
+  (void)filename;
+  AssertThrow(false, ExcMessage ("HDF5 support is disabled."));
+#  else
+
+  /*
+   * The content of the distributed matrix is copied to a matrix using a 1x1 process grid.
+   * Therefore, one process has all the data and can write it to a file.
+   */
+  //create a 1x1 column grid with P being the number of MPI processes
```

Comment:
~~with P being the number of MPI processes~~ (as P is not mentioned here anyway). 

Maybe also add

... column grid which will be used to initialize an effectively serial ScaLAPACK matrix to gather the contents from the current object